src/me/mrCookieSlime/Slimefun/Android/ProgrammableAndroid.java

```diff
@@ -1,1433 +1,1433 @@
+		blockblacklist.add(Material.STRUCTURE_BLOCK);
+	}
+
+	private Set<MachineFuel> recipes = new HashSet<>();
+
+	public String getInventoryTitle() {
+		return "Programmable Android";
+	}
+
+	public int[] getOutputSlots() {
+		return new int[] {20, 21, 22, 29, 30, 31};
+	}
+
+	public abstract AndroidType getAndroidType();
+	public abstract float getFuelEfficiency();
+	public abstract int getTier();
+
+	public ProgrammableAndroid(Category category, ItemStack item, String name, RecipeType recipeType, ItemStack[] recipe) {
+		super(category, item, name, recipeType, recipe);
+
+		if (getTier() == 1) {
+			registerFuel(new MachineFuel(80, new ItemStack(Material.COAL)));
+			registerFuel(new MachineFuel(80, new ItemStack(Material.CHARCOAL)));
+			registerFuel(new MachineFuel(800, new ItemStack(Material.COAL_BLOCK)));
+			registerFuel(new MachineFuel(45, new ItemStack(Material.BLAZE_ROD)));
+
+			// Logs
+			registerFuel(new MachineFuel(4, new ItemStack(Material.OAK_LOG)));
+			registerFuel(new MachineFuel(4, new ItemStack(Material.BIRCH_LOG)));
+			registerFuel(new MachineFuel(4, new ItemStack(Material.SPRUCE_LOG)));
+			registerFuel(new MachineFuel(4, new ItemStack(Material.JUNGLE_LOG)));
+			registerFuel(new MachineFuel(4, new ItemStack(Material.DARK_OAK_LOG)));
+			registerFuel(new MachineFuel(4, new ItemStack(Material.ACACIA_LOG)));
+
+			// Wooden Planks
+			registerFuel(new MachineFuel(1, new ItemStack(Material.OAK_PLANKS)));
+			registerFuel(new MachineFuel(1, new ItemStack(Material.BIRCH_PLANKS)));
+			registerFuel(new MachineFuel(1, new ItemStack(Material.SPRUCE_PLANKS)));
+			registerFuel(new MachineFuel(1, new ItemStack(Material.JUNGLE_PLANKS)));
+			registerFuel(new MachineFuel(1, new ItemStack(Material.DARK_OAK_PLANKS)));
+			registerFuel(new MachineFuel(1, new ItemStack(Material.ACACIA_PLANKS)));
+		}
+		else if (getTier() == 2){
+			registerFuel(new MachineFuel(100, new ItemStack(Material.LAVA_BUCKET)));
+			registerFuel(new MachineFuel(200, SlimefunItems.BUCKET_OF_OIL));
+			registerFuel(new MachineFuel(500, SlimefunItems.BUCKET_OF_FUEL));
+		}
+		else {
+			registerFuel(new MachineFuel(2500, SlimefunItems.URANIUM));
+			registerFuel(new MachineFuel(1200, SlimefunItems.NEPTUNIUM));
+			registerFuel(new MachineFuel(3000, SlimefunItems.BOOSTED_URANIUM));
+		}
+
+		new BlockMenuPreset(name, getInventoryTitle()) {
+
+			@Override
+			public void init() {
+				try {
+					constructMenu(this);
+				} catch (Exception e) {
+					e.printStackTrace();
+				}
+			}
+
+			@Override
+			public boolean canOpen(Block b, Player p) {
+				boolean open = BlockStorage.getLocationInfo(b.getLocation(), "owner").equals(p.getUniqueId().toString()) || p.hasPermission("slimefun.android.bypass");
+				if (!open) {
+					Messages.local.sendTranslation(p, "inventory.no-access", true);
+				}
+				return open;
+			}
+
+			@Override
+			public void newInstance(BlockMenu menu, final Block b) {
+				try {
+					menu.replaceExistingItem(15, new
```

Comment:
This check might as well be removed. It shouldn't even be here, must have slipped through my code review in an earlier PR.
You can remove this line I think.

Reply:
Yeah I was debating removing it because i didn’t think the code would ever even get that far if it wasn’t in block storage.